core/tee/tee_svc_storage.c

```diff
@@ -1047,18 +1196,28 @@ TEE_Result tee_svc_storage_obj_write(uint32_t obj, void *data, size_t len)
 
 	res = tee_ta_get_current_session(&sess);
 	if (res != TEE_SUCCESS)
-		return res;
+		goto exit;
 
 	res = tee_obj_get(sess->ctx, obj, &o);
 	if (res != TEE_SUCCESS)
-		return res;
+		goto exit;
 
-	if (!(o->flags & TEE_DATA_FLAG_ACCESS_WRITE))
-		return TEE_ERROR_ACCESS_CONFLICT;
+	if (!(o->info.handleFlags & TEE_HANDLE_FLAG_PERSISTENT)) {
+		res = TEE_ERROR_BAD_STATE;
+		goto exit;
+	}
+
+	if (!(o->flags & TEE_DATA_FLAG_ACCESS_WRITE)) {
+		res = TEE_ERROR_ACCESS_CONFLICT;
+		goto exit;
+	}
+
+	res = tee_obj_verify(sess, o);
```

Comment:
This is not necessary, the corrupt will be detected when the following tee_file_ops.write